File: Week6/ProblemSet6/sentimental-readability/readability.py
from sys import exit

sentence_detect = ['.', '!', '?']
word_detect = ' '
full_detection = [' ', ',', '.', '!', '?', ';', "'", '-']
sentence = input("Text: ")


# Counts letters, words and sentences
letter_count = 0
word_count = 1  # To include the last word in countiging
sentence_count = 0
for i in sentence:
    if i not in full_detection:
        # Assumes i is a letter
        letter_count += 1
    elif i in word_detect:
        word_count += 1
    elif i in sentence_detect:
        sentence_count += 1


# Get the average
average_letters = (letter_count / word_count) * 100
average_words = (sentence_count / word_count) * 100

# print(f"{letter_count}, {word_count}, {sentence_count}")
# print(f"{average_letters}, {average_words}")

# Coleman Formula
index = round(0.0588 * average_letters - 0.296 * average_words - 15.8)
if (index <= 1):
    print(f"Before Grade 1")
    exit()
elif (index >= 16):
    index = "16+"
print(f"Grade {index}")

File: Week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>


int str_length(char str[])
{
    int len = 0;
    for (int i = 0; str[i] != '\0'; i++)
    {
        len++;
    }
    return len;
}


int count_sentences(char str[])
{
    int len = str_length(str);
    int n_sentences = 0;
    char c;  // c means character

    for (int i = 0; i < len; i++)
    {
        c = str[i];
        if ((c == '.') || (c == '!') || (c == '?'))
        {
            n_sentences++;
        }
    }
    return n_sentences;
}


int count_words(char str[])
{
    int len = str_length(str);
    int n_words = 0;
    char c;  // c means character

    // Start counting words
    for (int i = 0; i < len; i++)
    {
        c = str[i];
        if (c == ' ')
        {
            n_words++;
        }
    }
    // Word at the end of the sentence isn't counted on for loop
    n_words++;  // so we count it manually;
    return n_words;
}


int count_letters(char str[])
{
    int len = str_length(str);
    int n_letters = 0;
    char c;  // c means character

    // Start counting letters
    for (int i = 0; i < len; i++)
    {
        c = str[i];
        if (isalpha(c) != 0)
        {
            n_letters++;
        }
    }
    return n_letters;
}


int main(void)
{
    string inputted_paragraph = get_string("Text: ");

    float n_letters = count_letters(inputted_paragraph);
    float n_words = count_words(inputted_paragraph);
    float n_sentences = count_sentences(inputted_paragraph);

    // Coleman-Liau
    float L = (n_letters / n_words) * 100.0;  // average number of letters per 100 words
    float S = (n_sentences / n_words) * 100.0;  // average number of sentences per 100 words

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}